jwst/pipeline/calwebb_spec2.py

```diff
@@ -43,179 +48,226 @@ class Spec2Pipeline(Pipeline):
-                log.error('No output product will be created')
-                continue
-
-            # Do background processing, if necessary
-            if 'bkgexps' in member and len(member['bkgexps']) > 0:
-
-                # Assemble the list of background exposures to use
-                bkg_list = []
-                for bkg in member['bkgexps']:
-                    bkg_list.append(bkg['expname'])
-
-                # Call the background subtraction step
-                input = self.bkg_subtract(input, bkg_list)
-
-                # Save the background-subtracted product, if requested
-                if self.save_bsub:
-                    if isinstance(input, datamodels.CubeModel):
-                        self.save_model(input, "bsubints")
-                    else:
-                        self.save_model(input, "bsub")
-
-            # Apply NIRSpec MSA imprint subtraction
-            if exp_type in ['NRS_MSASPEC', 'NRS_IFU']:
-                if 'imprint' in member and len(member['imprint']) > 0:
-                    imprint_filename = member['imprint'][0]['expname']
-                    input = self.imprint_subtract(input, imprint_filename)
-
-            # Apply NIRSpec MSA bad shutter flagging
-            # Stubbed out as placeholder until step module is created
-            #if exp_type in ['NRS_MSASPEC', 'NRS_IFU']:
-            #    input = self.msa_flagging(input)
-
-            # Extract 2D sub-windows for NIRSpec slit and MSA
-            if exp_type in ['NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ', 'NRS_MSASPEC']:
-                input = self.extract_2d(input)
-
-            # Apply flat-field correction
-            input = self.flat_field(input)
-
-            # Apply the source type decision step
-            input = self.srctype(input)
-
-            # Apply the straylight correction for MIRI MRS
-            if exp_type == 'MIR_MRS':
-                input = self.straylight(input)
-
-            # Apply the fringe correction for MIRI MRS
-            if exp_type == 'MIR_MRS':
-                input = self.fringe(input)
-
-            # Apply pathloss correction to NIRSpec exposures
-            if exp_type in ['NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ', 'NRS_MSASPEC',
-                            'NRS_IFU']:
-                input = self.pathloss(input)
-
-            # Apply flux calibration
-            input = self.photom(input)
-
-            # Record ASN pool and table names in output
-            if input_table.poolname:
-                input.meta.asn.pool_name = input_table.poolname
-                input.meta.asn.table_name = input_table.filename
-            else:
-                input.meta.asn.pool_name = ' '
-                input.meta.asn.table_name = ' '
+        asn = Level2Association.open(input)
+
+        # Each exposure is a product in the association.
+        # Process each exposure.
+        for product in asn['products']:
+            log.info('Processing exposure {}'.format(product['name']))
```

Comment:
Nomenclature question: We are technically processing individual exposures at level-2b, rather than a collection of exposures, but the name that's going to appear in this log line will be the name of the output product that's in the process of being created, so shouldn't we perhaps change the message to read "Processing **product** ..."